src/frontend/hooks/blog-form.js

```diff
@@ -0,0 +1,201 @@
+    if (link === "") { tmpErrors.push("link cannot be blank") }
+    if (date === "") { tmpErrors.push("date cannot be blank") }
+    if (category === "") {  tmpErrors.push("category cannot be blank") }
+
+    if (currentImageURL == '' && uploadedImageURL == '') {
+          tmpErrors.push( "One image is required")
+        }
+
+    const pattern = /^(([0-9])|([0-2][0-9])|([3][0-1]))\-(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)\-\d{4}$/;
+    if (!pattern.test(date)){
+      tmpErrors.push("date must be in day-month-year form")
+    }
+
+    if (summary.length > 200 || summary.length < 1){
+      tmpErrors.push(`Summary must be between 1-200 characters, currently: ${summary.length}`)
+    }
+
+    setErrors(tmpErrors);
+
+    if (tmpErrors.length == 0) {
+      return true; // checks passed
+    }
+    return false // cheks failed
+  }
+
+  // Begin submitting form
+  const saveForm = useCallback(async () => {
+    // TODO: Still need to validate whether someone uploaded a file that already exists in google storage
+    if(!allFieldsAreFilled()) return;
+
+    try {
+      if (uploadedImageURL != '') {
+        // Send new image to google storage
+        const file = uploadedImageURL;
+
+        const data = new FormData();
+        data.append('files', file, file.name);
+
+        const res = await api.formUpload(data);
+
+        const imageUrls = res.data.data;
+        if (imageUrls.length === 0) {
+          throw new Error('Failed to upload image.');
+        }
+      }
+
+      // Prepare image url to store in DB
+      const googleStorageImageLink = "https://storage.googleapis.com/waterloop_cms_image_upload/"
+       // Either keep the current image or save the name of the uploaded file (date-name)
+      const imageToUpload = uploadedImageURL == '' ? currentImageURL : `${googleStorageImageLink}${Date.now()}-${uploadedImageURL.name.replaceAll(' ', '-')}`
+
+      const blogData = {
+        title: blogTitle,
+        author,
+        summary,
+        date,
+        link,
+        closed,
+        visibility,
+        category,
+        image: imageToUpload
+      };
+
+      // Send to DB
+      if (allFieldsAreFilled()) {
+        if (params.blogId) {
+          // update an EXISTING blog (by id) with the new blogData we just defined
+          const res = await api.blogs.editBlog(params.blogId, blogData);
+          const data = await res.data
+          console.log("Tried to edit blog, status: " , data)
+        } else {
+          // add a NEW blog (no id) with the new blogData we just defined
+          const res = await api.blogs.addBlog(blogData);
+          const data = await res.data
+          console.log("Tried to add a new blog, status: " , data)
+        }
+      }
+
+      // onSuccess:
+      closeForm();
+    } catch (e) {
+      console.log("could not edit / add blog")
+      console.error(e);
+      setErrors(["Upload Failed; Please Refresh And Try Again."]);
```

Comment:
Just double-checking, did you test to make sure that if an error code is returned, then the program will hop into this `catch` block ?

Reply:
I feel like I did ... but it was pretty long ago so I think I should reconfirm